integration-tests/client/test/utils.ts

```diff
@@ -229,3 +229,79 @@ export async function moduleAddress(
     ).account as ModuleAccount
   ).baseAccount?.address!;
 }
+
+/**
+ * Generic function to wait for a state change by retrying a function until condition is met
+ */
+export async function waitForStateChange<T>(
+    checkFunction: () => Promise<T>,
+    condition: (result: T) => boolean,
+    maxAttempts: number = 60,
+    intervalMs: number = 500,
+    timeoutErrorMessage: string = "Timed out waiting for state change"
+): Promise<T> {
+  let attempts = 0;
+
+  while (attempts < maxAttempts) {
+    const result = await checkFunction();
+
+    if (condition(result)) {
+      return result;
+    }
+
+    attempts++;
+    await sleep(intervalMs);
+  }
+
+  throw new Error(`${timeoutErrorMessage} after ${maxAttempts} attempts`);
+}
+
+/**
+ * Wait for a balance to change (increase from initial value)
+ */
+export async function waitForBalanceChange({
+                                             client,
+                                             address,
+                                             denom,
+                                             initialBalance,
+                                             maxAttempts = 60,
+                                             intervalMs = 500,
+                                           }: {
+  client: StrideClient | CosmosClient;
+  address: string;
+  denom: string;
+  initialBalance: string;
```

Comment:
nit: you can probably remove initialBalance and just get the initial balance in the function itself

Reply:
done. thank you for spotting that.